opqueue.go

```diff
@@ -29,49 +29,41 @@ var (
 // It blocks on dequeue if the queue is empty, but returns an error if the
 // queue is full during enqueue.
 type OpQueue struct {
-	cond *sync.Cond
-	ctx  context.Context
-	can  context.CancelFunc
-
+	mu      sync.Mutex
+	cond    sync.Cond
 	depth   int
 	width   int
 	q       *list.List
 	entries map[ID]*OpSet
+	closed  bool
 }
 
 // NewOpQueue create a new OpQueue.
 func NewOpQueue(depth, width int) *OpQueue {
```

Comment:
optionally NewOpQueue could take a context, and we could pass in the actor context instead of background